src/nagini_translation/translators/statement.py

```diff
@@ -727,6 +727,16 @@ def translate_stmt_For(self, node: ast.For, ctx: Context) -> List[Stmt]:
                                                   None, None, target_type,
                                                   node, ctx)
 
+        cond = self.viper.EqCmp(err_var.ref(),
```

Comment:
Just for curiosity, could you explain what was the problem with for loops, please?

Reply:
This: https://github.com/marcoeilers/nagini/issues/115
Basically, Nagini generated code for for-loops that isn't well-defined when the container one iterates over is empty (because it talks about the contents of the container).